app/helpers/application_helper.rb

```diff
@@ -278,4 +278,9 @@ def sign_up_path
       new_user_registration_path
     end
   end
+
+  def unread_notifications(user)
+    available_conferences_ids = Conference.with_roles([:admin, :organizer, :cfp], user).pluck(:id)
+    Comment.find_since_last_login(user).where(commentable_type: 'Event', commentable_id: Event.where(conference_id: available_conferences_ids))
```

Comment:
I think that here you could simply use Comment.accessible_by(current_ability) which would make all this much more understandable.